src/rwf_sonar.cpp

```diff
@@ -7,64 +7,66 @@
 #include <cmath>
 #include <vector>
 
-class rwf {
+class RightWallFollow {
 
 public:
-
     // Class constructor
-    rwf(ros::NodeHandle n);
+    RightWallFollow(ros::NodeHandle n);
+
+    // Right wall follow algorithm
+    void move();
+	
+private:
+    RightWallFollow() = default;
 
     // Action message 
-	geometry_msgs::Twist action;
+    geometry_msgs::Twist action;
 
     // ROS necessities
     ros::Publisher pub;
     ros::Subscriber reading_front;
     ros::Subscriber reading_right;
 
     // Threshold where certain actions are taken (in cm)
-    float right_thresh = 23;
-    float front_thresh = 23;
+    float right_thresh = 23.f;
+    float front_thresh = 23.f;
 
     // Sensor readings
-	float front_range;
-    float right_range;
+    float front_range = 0.0f;
+    float right_range = 0.0f;
 
     // Callback functons to get readings from front and right sensors
-	void RangeScanCallbackFrontSensor(const sensor_msgs::Range::ConstPtr& frontReading);
-    void RangeScanCallbackRightSensor(const sensor_msgs::Range::ConstPtr& rightReading);    
-
-    // Right wall follow algorithm
-	void move();
-
+    void rangeScanCallbackFrontSensor(const sensor_msgs::Range::ConstPtr& frontReading);
+    void rangeScanCallbackRightSensor(const sensor_msgs::Range::ConstPtr& rightReading);    
 };
 
-rwf::rwf(ros::NodeHandle n){
-
+RightWallFollow::RightWallFollow(ros::NodeHandle n) :
+    RightWallFollow{}
+{
     ROS_INFO("Initializing right wall follow object");
 
-    reading_front = n.subscribe("robot0/sonar_0", 1000, &rwf::RangeScanCallbackFrontSensor,this);
+    reading_front = n.subscribe("robot0/sonar_0", 1000, &rwf::rangeScanCallbackFrontSensor,this);
```

Comment:
Maybe load topic names from parameter server with a default value.
Same for queue size. 1000 also seems big, but idk how fast the sensors are shitting out data,